#! /bin/sh /usr/share/dpatch/dpatch-run
## collectd_memleak.dpatch by Peter Holik <holik@it-technology.at>
##
## DP: Free allocated memory before returning in an error condition.
## DP: (This is upstream Git commit 111a5eb7b68da0478b16de31dff422a213a03799)

@DPATCH@

diff a/src/utils_threshold.c b/src/utils_threshold.c
--- a/src/utils_threshold.c
+++ b/src/utils_threshold.c
@@ -711,6 +711,7 @@ int ut_check_interesting (const char *name)
   if (status != 0)
   {
     ERROR ("ut_check_interesting: parse_identifier failed.");
+    sfree (name_copy);
     return (-1);
   }
 
